File: app/src/main/java/ru/mamykin/foreignbooksreader/di/modules/PreferencesModule.java
package ru.mamykin.foreignbooksreader.di.modules;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import ru.mamykin.foreignbooksreader.preferences.PreferencesManager;

/**
 * Creation date: 5/29/2017
 * Creation time: 11:39 AM
 * @author Andrey Mamykin(mamykin_av)
 */
@Module
public class PreferencesModule {
    @Provides
    @Singleton
    PreferencesManager providePreferencesManager() {
        return new PreferencesManager();
    }
}

File: app/src/main/java/ru/mamykin/foreignbooksreader/di/modules/PreferencesModule.kt
package ru.mamykin.foreignbooksreader.di.modules

import javax.inject.Singleton

import dagger.Module
import dagger.Provides
import ru.mamykin.foreignbooksreader.preferences.PreferencesManager

/**
 * Creation date: 5/29/2017
 * Creation time: 11:39 AM
 * @author Andrey Mamykin(mamykin_av)
 */
@Module
class PreferencesModule {
    @Provides
    @Singleton
    internal fun providePreferencesManager(): PreferencesManager {
        return PreferencesManager()
    }
}